compiler/optimizer/OMRCFGSimplifier.cpp

```diff
@@ -333,6 +333,276 @@ bool OMR::CFGSimplifier::simplifyInstanceOfTestToCheckcast(bool needToDuplicateT
+
+      if (firstChild->getOpCodeValue() == TR::iload) 
+         {
+         if (secondChild->getOpCodeValue() == TR::iconst && secondChild->getConstValue() == 0) 
+            {
+            patternMatched = true;
+            throwBlock = isCompareLT ? compareNode->getBranchDestination()->getNode()->getBlock() : _block->getNextBlock();
+            nextCompareBlock = isCompareLT ? _block->getNextBlock() : compareNode->getBranchDestination()->getNode()->getBlock();
+            firstChildBndChk = compareNode->getFirstChild();
+           } else if (secondChild->getOpCode().isIntegralLoadVar() || secondChild->getOpCode().isLoadIndirect()) {
+            patternMatched = true;
+            throwBlock = isCompareLT ? _block->getNextBlock() : compareNode->getBranchDestination()->getNode()->getBlock();
+            nextCompareBlock = isCompareLT ? compareNode->getBranchDestination()->getNode()->getBlock() : _block->getNextBlock();
+            secondChildBndChk = compareNode->getSecondChild();
+         }
+      }
+   }
+
+   if (!patternMatched)
+      return false;
+
+   if (trace())
+      traceMsg(comp(), "   Matched with first %s in block_%d\n", isCompareLT ? "ificmplt" : "ificmpge",
+                  _block->getNumber());
+
+   TR::Node *secondCompareNode = nextCompareBlock->getLastRealTreeTop()->getNode();
+
+   if (secondCompareNode->getOpCodeValue() != TR::ificmplt && secondCompareNode->getOpCodeValue() != TR::ificmpge) 
+      {
+      if (trace())
+         traceMsg(comp(), "   Abort simplifyBoundCheckWithThrowException : The second compare block does not contain ificmplt or ificmpge\n");
+      return false;
+      }
+
+   bool firstNodeRequiresDuplicate = false;
+   bool secondNodeRequiresDuplicate = false;
+   bool isLastBlockRetBlock = true;
+   patternMatched = false;
+   TR::Block *retBlock = NULL;
+   TR::Block *secondThrowBlock = NULL;
+   bool isSecondCompareLT = secondCompareNode->getOpCodeValue() == TR::ificmplt;
+
+   if (secondCompareNode->getNumChildren() > 1) 
+      {
+      TR::Node *firstChild = secondCompareNode->getFirstChild();
+      TR::Node *secondChild = secondCompareNode->getSecondChild();
+      
+      if (firstChild->getOpCodeValue() == TR::iload) 
+         {
+         if (secondChild->getOpCodeValue() == TR::iconst && secondChild->getConstValue() == 0) 
+            {
+            patternMatched = true;
+            secondThrowBlock = isSecondCompareLT ? secondCompareNode->getBranchDestination()->getNode()->getBlock() : nextCompareBlock->getNextBlock();
+            retBlock = isSecondCompareLT ? nextCompareBlock->getNextBlock() : secondCompareNode->getBranchDestination()->getNode()->getBlock();
+            isLastBlockRetBlock = !isSecondCompareLT;
+            firstChildBndChk = secondCompareNode->getFirstChild();
+            firstNodeRequiresDuplicate = true;
+         } else if (secondChild->getOpCode().isIntegralLoadVar() || secondChild->getOpCode().isLoadIndirect()) {
```

Comment:
the brace formatting here does not conform to the rest of the code

Reply:
Fixed